procmonAnalyze.js

```diff
@@ -1429,12 +1655,78 @@ function handleMouseUp(event) {
     if (event.which == 2 || event.button == 4 ) {
       event.preventDefault();
       gState.middleMouseDownFor = null;
-    } else {
+    } else if (event.explicitOriginalTarget.id == "redraw") {
+      event.preventDefault();
+      refocusArea();
+    } else if (event.explicitOriginalTarget.id == "reset") {
+      event.preventDefault();
+      resetData();
+    } else if (event.explicitOriginalTarget.id == "canvas-markers") {
+      event.preventDefault();
+      isDownTopSlider = false;
+      isDownBottomSlider = false;
+    }
+  }
+}
 
+function handleMouseUpOut(event) {
+  event.preventDefault();
+  isDownTopSlider = false;
+  isDownBottomSlider = false;
+}
+
+/*
+  Focuses the canvas on the specified range.
+*/
+function refocusArea() {
+  if (gState) {
+    let constrainedData = gOuterState.unconstrainedData.filter(row => 
+      row.start > gOuterState.constrainedTimeBegin && row.start + row.duration < gOuterState.constrainedTimeEnd);
+    // TODO: constrain diskify data
+    if (constrainedData) {
+      //TO DO: this should have diskify data
+      drawData(constrainedData, null, true);
     }
   }
 }
 
+/*
+  Returns the data back to the original view.
+*/
+function resetData() {
+  if (gState) {
+    drawData(data, diskifyData, false);
+  }
+}
+
+function absTimeToRelTime(absTime) {
```

Comment:
Relative time, as the code has been informally treating it prior to this, was always relative to the `minTime` as defined by `gState`. If we use `gOuterState` here, then we're going to run into problems, and some of the bugs I'm running into feel like that's what's going on?